### PR TITLE
dev/core#3495 - Fix fields with wildcards in Advanced Search

### PR DESCRIPTION
**Pull request by bsilvern**
Overview
----------------------------------------
In the affected function, some fields are arrays, e.g. ['LIKE' => 'field-value'], rather than strings as was expected. Modified to now check for that and handle correctly.

Before
----------------------------------------
1. Some search terms, e.g. Contribution Source, are changed to '%%' upon sorting or paging the results.
2. After fixing the above, an extraneous trailing '%' on search terms was appending on each sort of paging of the results.

After
----------------------------------------
1. All search terms are preserved.
2. Extraneous wildcards are no longer appended.

Technical Details
----------------------------------------

Comments
----------------------------------------

### CODE REVIEW

**Comment by seamuslee001** [comment]
Jenkins ok to test

File: CRM/Contact/BAO/Query.php [hunk 1]
```diff
@@ -6309,15 +6309,13 @@ public static function getQillValue($daoName, string $name, $value, $op, string
   public static function getWildCardedValue($wildcard, $op, $value) {
     if ($wildcard && $op === 'LIKE') {
       if (CRM_Core_Config::singleton()->includeWildCardInName && (substr($value, 0, 1) != '%')) {
-        return "%$value%";
+        $value = "%$value";
```
Comment (totten) [inline CRM/Contact/BAO/Query.php:6312]: Before, it wrapped the string with `%`s on both sides. After, it only prepends `%`.  

My memory of this codepath is cold -- but wouldn't that be a substantive change in the search behavior?

Comment (bsilvern) [reply CRM/Contact/BAO/Query.php:6312]: See the following two lines (6314-6315) which add the trailing '%' if it does not already exist.

**Comment by eileenmcnaughton** [comment]
I was able to replicate this and confirm that the patch fixes the bug and that paging still works on the Contribution Search (which did not have this bug).

Much as I hate touching anything at all in the legacy search code now this is clearly a bug rather than a new feature and stepping through the code I could see it was carefully implemented with handling for both forms.

I'm going with 'gratefully accept this patch'

**Pull request merged**